cmd/boskos/boskos.go

```diff
@@ -144,3 +172,69 @@ func main() {
 	// signal to the world that we're ready
 	health.ServeReady()
 }
+
+type configSyncReconciler struct {
+	sync func() error
+}
+
+func (r *configSyncReconciler) Reconcile(_ reconcile.Request) (reconcile.Result, error) {
+	err := r.sync()
+	if err != nil {
+		logrus.WithError(err).Error("Config sync failed")
+	}
+	return reconcile.Result{}, err
+}
+
+func addConfigSyncReconcilerToManager(mgr manager.Manager, configSync func() error, configChangeEvent <-chan event.GenericEvent) error {
+	ctrl, err := controller.New("bokos_config_reconciler", mgr, controller.Options{
+		// We reconcile the whole config, hence this is not safe to run concurrently
+		MaxConcurrentReconciles: 1,
+		Reconciler: &configSyncReconciler{
+			sync: configSync,
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to construct controller: %w", err)
+	}
+
+	if err := ctrl.Watch(&source.Kind{Type: &crds.ResourceObject{}}, constHandler(), resourceUpdatePredicate()); err != nil {
+		return fmt.Errorf("failed to watch boskos resources: %w", err)
+	}
+	if err := ctrl.Watch(&source.Kind{Type: &crds.DRLCObject{}}, constHandler()); err != nil {
+		return fmt.Errorf("failed to watch boskos dyanmic resources: %w", err)
+	}
+	if err := ctrl.Watch(&source.Channel{Source: configChangeEvent}, constHandler()); err != nil {
+		return fmt.Errorf("failed to create source channel for config change event: %w", err)
+	}
+	if err := mgr.Add(ctrl); err != nil {
+		return fmt.Errorf("failed to add controller to manager: %w", err)
+	}
+
+	return nil
+}
+
+func constHandler() handler.EventHandler {
+	return &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(func(handler.MapObject) []reconcile.Request {
+			return []reconcile.Request{{}}
+		}),
+	}
+}
+
+// resourceUpdatePredicate prevents the config reconciler from reacting to resource update events
```

Comment:
it doesn't need to happen in this PR, but it seems like we might be able to tweak this to fix https://github.com/kubernetes-sigs/boskos/issues/20 as well.

Reply:
Added that